manopt/tools/checkdiff.m

```diff
@@ -114,5 +114,7 @@ function checkdiff(problem, x, d, force_gradient)
     fprintf('The slope should be 2. It appears to be: %g.\n', poly(1));
     fprintf(['If it is far from 2, then directional derivatives ' ...
              'might be erroneous.\n']);
+    fprintf(['If it is very close to 0, then the modeled funtion is '...
```

Comment:
Since this scenario is quite rare, could you please place this fprintf(..) inside and if .. end statement, so that it is displayed only if there is a chance that it will be relevant for the user?

Also: is it possible that the slope would turn out to be NaN instead of 0 in the targeted case?